R/Lithuania.R

```diff
@@ -0,0 +1,469 @@
+                .data$ab_pos_day / .data$ab_tot_day
+              ),
+            ag_prc_day =
+              dplyr::if_else(
+                .data$ag_tot_day == 0, 0,
+                .data$ag_pos_day / .data$ag_tot_day
+              ),
+            pcr_prc_day =
+              dplyr::if_else(
+                .data$pcr_tot_day == 0, 0,
+                .data$pcr_pos_day / .data$pcr_tot_day
+              ),
+            dgn_prc_day =
+              dplyr::if_else(
+                .data$dgn_tot_day == 0, 0,
+                .data$dgn_pos_day / .data$dgn_tot_day
+              )
+          )
+      }
+    },
+
+    #' @description Lithuania Specific Municipality Level Data Cleaning
+    #'
+    #' No data cleaning is necessary specific to the municipality level
+    #' since the data is already provided at the municipality level.
+    #'
+    #' @importFrom dplyr mutate group_by summarise ungroup full_join
+    #'
+    clean_level_2 = function() {
+    },
+
+    #' @description Initialize the country
+    #' @param death_definition A character string. Determines which criteria
+    #'   for attributing deaths to COVID is used. Should be `"of"`,
+    #'    `"with"`, or `"after"`. Can also be `"daily_deaths_def1"`,
+    #'    `"daily_deaths_def2"`, or `"daily_deaths_def3"`. (Defaults
+    #'    to `"of"`, the strictest definition.)
+    #' @param recovered_definition A character string. Determines whether
+    #'   the count of officially-recovered (*de jure*) cases is used, or
+    #'   the statistical estimate provided by OSP. Should be `"official"`
+    #'   or `"statistical"`. (Defaults to `"official"`.)
+    #' @param national_data A logical scalar. Should national values be
+    #'   returned?  (Defaults `FALSE`)
+    #' @param all_osp_fields A logical scalar. Should all the meaningful
+    #'   data fields from the OSP source be returned? (Defaults `FALSE`)
+    #'
+    #' @param ... The args passed by [general_init]
+    initialize = function(death_definition = "of",
+                          recovered_definition = "official",
+                          all_osp_fields = FALSE,
+                          national_data = FALSE, ...) {
+      general_init(self, ...)
+      # Add custom fields here
+      private$death_definition <- death_definition
+      private$recovered_definition <- recovered_definition
+      private$all_osp_fields <- all_osp_fields
+      private$national_data <- national_data
+    }
+  ),
+  private = list(
+    # death_definition which criteria of deaths attributed to
+    # COVID to use
+    death_definition = "of",
+    # recovered_definition whether to use the official counts of
+    # recovered cases or the statistical estimates provided by OSP
+    recovered_definition = "official",
+    # all_osp_fields whether to return all the data vectors provided
+    # by OSP
+    all_osp_fields = FALSE,
+    # national_data whether to return data rows for national results
+    national_data = FALSE
+  )
```

Comment:
I think we should make these public fields with documentation

Reply:
I will move these to public. It may not be clear, but the comments here used to be `@field` documentation then I learned that you're not supposed to document private fields.

Do we end up with duplicate documentation if these fields essentially are just the internal mirror of parameters? And is there any way to make them more visible in the documentation of `get_regional_data`?

Reply:
Yes unfortunately they do end up being repeated. Documenting for R6 classes gets tricky when it comes to inheritance so having a genral params gives an advantage of sharing code between countries but you do loose out on having the params explained in that countries documentation. The solution we have at the moment is to provide a link in the dot params argument which directs to the docs for `general_init()`.

Reply:
These shifts have now been made. I think I *may* put in an offer of the documentation (as in the current Lithuania code), suggesting that people do `?covidregionaldata::Lithuania` for more details on the data set.